### ISSUE TITLE
oauth2Provider properties listed as optional but are mandatory

### ISSUE THREAD

**Issue by macMikey** [4d/4D-NetKit#68]
* the **tokenURI** is not optional when building a jwt for google
(stay tuned, more to come, maybe)

**Comment by e-marchand** [comment]
`response_type` seems to be set as "code" , because it cannot be changed for the moment, I think it must be documented as something that you could pass

https://github.com/4d/4D-NetKit/blob/a77462e4704ec889fe8eee884e735fe8232b57db/Project/Sources/Classes/OAuth2Provider.4dm#L876


--

scope is not mandatory in Oauth2 protocol, it depend on api you use, it could even be ignored by server according to your right
the server must choose a default one (as RFC say)

**Comment by macMikey** [comment]
right. it's different, depending on the provider. rather than just changing the docs and issuing a PR, i wanted to throw out an issue, first, to discuss.
what i would suggest is that we add a column to show the differences between different consumers
i would also suggest that we sort the properties alphabetically in the list.
